docs/adrs/006-web-apis.md

```diff
@@ -0,0 +1,240 @@
+
+When developing a web application for communicating with the Penumbra blockchain, dApp makers might need the following features:
+
+- Get a list of injected wallets
+- Connect to account
+- Disconnect from account
+- Monitor the connection
+- Get and display private information about the account
+- Get and display public information about the blockchain
+- Get real-time updates about the account and new blocks
+- Sign and send transactions within Penumbra
+- Wait for transactions to be complete
+- Send IBC transactions
+- Estimate transaction costs
+- Stake and swap assets
+- Participate in protocol governance
+
+This list is supposed to be covered by the proposed API, documented and presented with examples to users.
+
+## Creating the client
+
+In other blockchains, JavaScript SDKs usually split the interfaces into at least two parts: one that manages the wallet connection and private transactions, and one that reads the public blockchain data. Examples include [Viem](https://viem.sh/docs/clients/intro) and [Thirdweb](https://portal.thirdweb.com/typescript/v5/client) on Ethereum.
+
+In Penumbra, any blockchain query requires a connection with the wallet provider, therefore all actions and
+queries can be called 'private'. It is reasonable to construct the notion of a **client** – the interface 
+that manages the injected connections and provides methods for interacting with the blockchain.
+
+Creating the `client` should be the starting point for any application working with Penumbra:
+
+```ts
+import { createPenumbraClient } from '@penumbra-zone/client';
+
+export const client = createPenumbraClient();
+```
+
+The flow of work with the `client` would be as follows:
+
+- User creates an instance of the `client`
+- They use the `client` to establish the injected connection, and the `client` saves the connection in its state
+- To fetch information from the wallet, the application passes the `client` instance to the fetch methods
+
+The idea of a single shared `client` is inspired by these popular TypeScript libraries:
+- [Prisma ORM](https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/instantiate-prisma-client)
+- [Supabase](https://supabase.com/docs/reference/javascript/initializing)
+- [Viem](https://viem.sh/docs/clients/intro)
+
+## Injected connection
+
+The `client` must be able to connect to injected wallets. To make the API possible, the interface injected into the `window` object must be standardized:
+
+```ts
+export interface PenumbraProvider extends Readonly<PenumbraStateEventTarget> {
+  /** Should contain a URI at the provider's origin, 
+   * serving a manifest describing this provider */
+  readonly manifest: string;
+
+  /** Call to gain approval. Returns `MessagePort` to this provider. 
+   * Might throw descriptive errors if user denies the connection */
+  readonly connect: () => Promise<MessagePort>;
```

Comment:
yeah, it combines both functions together

Reply:
previously these were combined, then separated because it was asked that they become separate. we've had recent discussions where it was suggested that helper functions exported from the client package should not helpfully perform this combination. so it seems like opinions are contradictory.

these are currently separate and i think should remain separate.

if they are combined, then a long 'connection init is happening' wait is indistinguishable from a long 'user approval is happening' wait. it is useful for these to be distinct time periods.

Reply:
In what cases would a consumer want to initiate a connection, but not trigger an approval? This behavior does not feel standard amongst wallet APIs in the industry.

Reply:
can't think of the cases when the `request` shouldn't be followed by `connect`. Don't know about the previous decisions but reverting them doesn't seem to be really breaking